test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml

```diff
@@ -67,6 +67,9 @@
         publisher: Canonical
         sku: 16.04-LTS
         version: latest
+      custom_data: |
+        #!/bin/sh
+        echo "custom_data was executed" > /custom_data.txt
   register: output
```

Comment:
Can you add a test to make sure this script actually executed?

Reply:
@mattclay Hi, I've expanded the tests as you requested. Would you have a moment to review them? Thanks! Happy New Year!